frontend/src/app/hierarchy-viewer/hierarchy-viewer.component.ts

```diff
@@ -19,32 +23,194 @@ export class HierarchyViewerComponent implements OnDestroy {
+
+                const fileReader = new FileReader();
+                fileReader.readAsText(importFile, "UTF-8");
+                fileReader.onload = () => {
+                    const hierarchyRequest: HierarchyRequest = JSON.parse(fileReader.result as string);
+                    this.selectedHierarchy = null;
+
+                    //The exported file contains ids and icons, so we need to fix that on import or export
+                    this.store.dispatch(createHierarchy({ hierarchy: hierarchyRequest }));
+                }
+                fileReader.onerror = (error) => {
+                    console.log(error);
+                }
+            }
+        });
+    }
+}
+
+@Component({
+    selector: 'create-hierarchy-dialog',
+    templateUrl: 'createHierarchy.dialog.html',
+    styleUrls: ['./hierarchy-viewer.component.scss']
+})
+export class CreateHierarchyDialog {
+    constructor(
+        public dialogRef: MatDialogRef<CreateHierarchyDialog>,
+        @Inject(MAT_DIALOG_DATA) public data: any) { }
+
+    doAction() {
+        this.dialogRef.close({ event: 'Create', name: this.data.name, description: this.data.description });
+    }
+
+    closeDialog() {
+        this.dialogRef.close({ event: 'Cancel' });
+    }
+}
+
+@Component({
+    selector: 'delete-hierarchy-dialog',
+    templateUrl: 'deleteHierarchy.dialog.html',
+    styleUrls: ['./hierarchy-viewer.component.scss']
+})
+export class DeleteHierarchyDialog {
+
+    constructor(
+        public dialogRef: MatDialogRef<DeleteHierarchyDialog>,
+        @Inject(MAT_DIALOG_DATA) public data: any) { }
+
+    doAction() {
+        this.dialogRef.close({ event: 'Delete' });
+    }
+
+    closeDialog() {
+        this.dialogRef.close({ event: 'Cancel' });
+    }
+}
+
+@Component({
+    selector: 'import-export-hierarchy-dialog',
+    templateUrl: 'importExportHierarchy.dialog.html',
+    styleUrls: ['./hierarchy-viewer.component.scss']
+})
+export class ImportExportHierarchyDialog {
+    tabs: string[] = ['Import', 'Export'];
+    selectedTabIndex: number = 0;
+    downloadJsonHref: SafeUrl | null = null;
+    downloadJsonName: string | null = null;
+    file: File | null = null;
+    fileName = '';
+
+    myTabSelectedIndexChange(index: number) {
+        this.selectedTabIndex = index;
+
+        // Current tab is export, downloadJsonHref has not been set and a hierarchy has been passed in
+        if (index == 1 && this.downloadJsonHref == null && this.data.hierarchy != null) {
+            this.generateDownloadJsonUri();
+        }
+    }
+
+    constructor(private sanitizer: DomSanitizer,
+        public dialogRef: MatDialogRef<ImportExportHierarchyDialog>,
+        @Inject(MAT_DIALOG_DATA) public data: any) {
+    }
+
+    onFileSelected(event: any) {
+        this.file = event.target.files[0];
+        this.fileName = this.file ? this.file.name : '';
+    }
+
+    generateDownloadJsonUri() {
+        var theJSON = JSON.stringify(this.data.hierarchy);
```

Comment:
It's better to use "let" rather than "var" because "let" puts variables in scope. If a variable isn't reassigned to use "const".